File: demo/src/main/kotlin/zx/base/s2/ktbase29.java
package zx.base.s2;
interface ResponseResut{
    void resule(String msg,int code);
}
public class ktbase29 {
    public static void main(String[] args) {
        loginApi("derry","dgdg", new ResponseResut() {
            @Override
            public void resule(String msg, int code) {
                System.out.println(msg+code);
            }
        });
    }

    private static void loginApi(String name, String password, ResponseResut responseResut) {
    if (name.equals("derry")&& password.equals("123456")){
        responseResut.resule("登录成功。",200);
    }else{
        responseResut.resule("登录失败。",444);
    }
    }


}

File: demo/src/main/kotlin/zx/base/s2/ktbase29.kt
package zx.base.s2

fun main() {

    //loginApi第三个参数是一个函数，可以提取到括号外面
    loginApi("derry","123456"){msg:String,code:Int->
        println("msg $msg ,code $code")

    }

}
//loginApi第三个参数是一个函数，类型是(String,Int)->Unit
fun loginApi(username:String ,password:String,responseResulet:(String,Int)->Unit){
    if (username == "derry" && password=="123456"){
        responseResulet("login success",200)
    }else{
        responseResulet("login fail",404)
    }}